astroid/inference.py

```diff
@@ -518,6 +519,8 @@ def _is_not_implemented(const):
 def _invoke_binop_inference(instance, opnode, op, other, context, method_name):
     """Invoke binary operation inference on the given instance."""
     methods = dunder_lookup.lookup(instance, method_name)
+    if context is not None:
```

Comment:
I think this could definitely be in another PR.

Reply:
I'll see if this passes any of the tests in a new PR. I think it depends on the explicit context